grails-app/controllers/org/pih/warehouse/core/BudgetCodeController.groovy

```diff
@@ -10,14 +10,21 @@
 package org.pih.warehouse.core
 
 class BudgetCodeController {
+    def budgetCodeService
 
     def index = {
         redirect(action: "list", params: params)
     }
 
     def list = {
         params.max = Math.min(params.max ? params.int('max') : 10, 100)
-        [budgetCodes: BudgetCode.list(params), budgetCodesTotal: BudgetCode.count()]
+
+        if (params.q) {
+            def budgetCodeList = budgetCodeService.findBudgetCodes(params.q)
+            return [budgetCodes: budgetCodeList, budgetCodesTotal: budgetCodeList.size()]
+        }
+
+        return [budgetCodes: BudgetCode.list(params), budgetCodesTotal: BudgetCode.count()]
```

Comment:
1. You could move this `if` directly to the `createCriteria`, so you would only have here:
```
List<BudgetCode> budgetCodes = budgetCodeService.list(params)
return [budgetCodes: budgesCodes, totalCount: budgetCodes.totalCount()]
```

2. Doing this:
```
budgetCodesTotal: budgetCodeList.size()
```
is wrong, because you won't get the `totalCount` for pagination by doing so, but only the size of actual results, so if your page has 10 results per page, the size would always be <= 10